web_programming/http_request_and_response.py

```diff
@@ -0,0 +1,69 @@
+import json
+import socket
+
+
+def request(
+    sock: socket.socket, req_type: str, endpoint: str, headers: list[str], body: str
+) -> None:
+    """
+    Request HTTP connection
+
+    :param sock: socket client
+    :param req_type: request type (ex: GET, POST)
+    :param endpoint: endpoint
+    :param headers: http headers
+    :param body: http body
```

Comment:
```suggestion
```
I'm not sure if these are very useful, the variable names should be self documenting enough